sbi/utils/vector_field_utils.py

```diff
@@ -22,3 +22,29 @@ def forward(self, theta: Tensor, x: Tensor, t: Tensor) -> Tensor:
             Vector field evaluation at the provided points
         """
         pass
+
+
+class MaskedVectorFieldNet(nn.Module, ABC):
+    """Abstract base class for vector field networks with masking support.
+
+    Used by models that require conditioning and edge masking, such as Siformer.
+    """
+
+    @abstractmethod
+    def forward(
+        self, inputs: Tensor, t: Tensor, condition_mask: Tensor, edge_mask: Tensor
```

Comment:
I would always support `edge_mask: Optional[Tensor]` i.e. `edge_mask=None` should work. (This should be the case by default as normally in transformers if you do not pass an mask it defaults to standard attention which is the same as a all ones edge_mask).